frameworks/keyed/ember/app/templates/components/my-table.hbs

```diff
@@ -1,73 +1,58 @@
+        </div>
+        <div class="col-sm-6 smallpad">
+          <BsButton id="runlots" @onClick={{action this.runLots}}>
+            Create 10,000 rows
+          </BsButton>
+        </div>
+        <div class="col-sm-6 smallpad">
+          <BsButton id="add" @onClick={{action this.add}}>
+            Append 1,000 rows
+          </BsButton>
         </div>
-        <div class="col-md-6">
-            <div class="row">
-                <div class="col-sm-6 smallpad">
-                    {{#bs-button id='run' onClick=(action create)}}
-                      Create 1,000 rows
-                    {{/bs-button}}
-                </div>
-                <div class="col-sm-6 smallpad">
-                    {{#bs-button id='runlots' onClick=(action runLots)}}
-                      Create 10,000 rows
-                    {{/bs-button}}
-                </div>
-                <div class="col-sm-6 smallpad">
-                    {{#bs-button id='add' onClick=(action add)}}
-                      Append 1,000 rows
-                    {{/bs-button}}
-                </div>
-                <div class="col-sm-6 smallpad">
-                    {{#bs-button id='update' onClick=(action update)}}
-                      Update every 10th row
-                    {{/bs-button}}
-                </div>
-                <div class="col-sm-6 smallpad">
-                    {{#bs-button id='clear' onClick=(action clear)}}
-                      Clear
-                    {{/bs-button}}
-                </div>
-                <div class="col-sm-6 smallpad">
-                    {{#bs-button id='swaprows' onClick=(action swapRows)}}
-                      Swap Rows
-                    {{/bs-button}}
-                </div>
-            </div>
+        <div class="col-sm-6 smallpad">
+          <BsButton id="update" @onClick={{action this.update}}>
+            Update every 10th row
+          </BsButton>
         </div>
+        <div class="col-sm-6 smallpad">
+          <BsButton id="clear" @onClick={{action this.clear}}>
+            Clear
+          </BsButton>
+        </div>
+        <div class="col-sm-6 smallpad">
+          <BsButton id="swaprows" @onClick={{action this.swapRows}}>
+            Swap Rows
+          </BsButton>
+        </div>
+      </div>
     </div>
+  </div>
 </div>
 
 
 <table class="table table-hover table-striped test-data">
   <tbody>
-  {{#each this.data key='id' as |item|}}
-
-    <tr class={{if (eq item.id this.selected) 'danger' 'x'}}>
-      <td class="col-md-1">{{item.id}}</td>
-
-      <td class="col-md-4">
-        <a onclick={{action this.select item.id}}>
-          {{item.label}}
-        </a>
-      </td>
-
-      <td class="col-md-1">
-
-        <a onclick={{action this.remove item.id}}>
-          <span class="glyphicon glyphicon-remove" aria-hidden="true"></span>
-        </a>
-      </td>
-
-      <td class="col-md-6"></td>
-    </tr>
-
-  {{/each}}
+    {{~#each this.data key="@index" as |item|~}}
```

Comment:
What is `@index` here, as it's not a prop passed to `{{my-table}}`! 🤔

Reply:
Found the docs on this! https://api.emberjs.com/ember/release/classes/Ember.Templates.helpers/methods/if?anchor=each

Reply:
though, more detail would be good there. I still don't know what all the options are for `key`, and why `@index` means the actual index. convention? hard coded?

Reply:
https://discuss.emberjs.com/t/new-glimmer-consideration-each-key/8128/9

Reply:
https://github.com/emberjs/ember.js/issues/16975

Reply:
I would have thought using the index as the key would result in stale / wrongly updated rows when things are removed / inserted.  🤷‍♂

Reply:
As I understand my goal is - get isolated component-based references https://github.com/glimmerjs/glimmer-vm/blob/master/guides/04-references.md

We have amazing kvo in ember/glimmer, and we need to use all benifits.
Previously ember project in this bench don't use `kvo` only mutable data, and glimmer-vm has only one valid reference (`id`), and it need to recalculate and revalidate all the stuff.
 
Now glimmer has index-like references to KVO objects, and it wins for cases where test suit change data inside rows.

I think @chancancode can clarify this from glimmer engineering side.

Reply:
> emberjs/ember.js#16975

That's interesting! So I have an idea now what `@index` means. But I still don't get why it made any difference in this case. 

The example in the linked issue was an array of primitive values, but in our case we have objects (EmberObjects now). So I would expect to having no `key` (same as `@identity) to just work the same (now that we mutate the items the identity stays the same). Or - as it was before - to have `key='id'`, which would also correctly match the items with their existing DOM elements.

So why is `@index` any better here?

Reply:
Have no idea( I have screenshots with `@index`/`@identity` and looks like somehow for this case `@index` more performant, glimmer https://github.com/krausest/js-framework-benchmark/pull/555/files works well with default `@identity` (can't get perf impact with `@index`)

We can install DOM mutation observers and measure mutation count for both cases

Reply:
![image](https://user-images.githubusercontent.com/1360552/56853816-79d9a280-6935-11e9-85eb-58773c8b39b5.png)

Reply:
on 30 runs perf close to preact
![image](https://user-images.githubusercontent.com/1360552/56853941-6cbdb300-6937-11e9-9486-648eac905cce.png)